runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/StateSpecFunctions.java

```diff
@@ -177,8 +185,14 @@
 
           // close and checkpoint reader.
           reader.close();
-          LOG.info("Source id {} spent {} msec on reading.", microbatchSource.getId(),
-              stopwatch.stop().elapsed(TimeUnit.MILLISECONDS));
+          final long readDurationMillis = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+
+          readDuration.set(readDurationMillis);
```

Comment:
Setting the value here will have splits overwriting each other. The real number we're looking for here is the max read duration among all splits per microbatch. Consider calculating this number in `ReadReportDStream`.